Module not found: Can't resolve 'react-native'

Since upgrading to `@aws-amplify/auth` from `1.2.15` to `1.2.17`, I'm getting the following error from `eslint`:

```
./node_modules/@aws-amplify/auth/lib/Auth.js
Module not found: Can't resolve 'react-native' in '/opt/atlassian/pipelines/agent/build/node_modules/@aws-amplify/auth/lib'
```

I'm getting this too .. and, I'm trying to create an ionic app .. not react.

`[ng] WARNING in ./node_modules/@aws-amplify/auth/lib/Auth.js
[ng] Module not found: Error: Can't resolve 'react-native' in node_modules\@aws-amplify\auth\lib'
`

--- a/packages/auth/package.json
+++ b/packages/auth/package.json
@@ -57,8 +57,7 @@
     "amazon-cognito-identity-js": "3.0.9-unstable.0"
   },
   "peerDependencies": {
-    "buffer": "4.x",
-    "react-native": "^0.44.0"
+    "buffer": "4.x"
   },
   "jest": {
     "transform": {

--- a/packages/auth/src/Auth.ts
+++ b/packages/auth/src/Auth.ts
@@ -52,40 +52,11 @@ import {
 } from 'amazon-cognito-identity-js';
 import { CognitoAuth } from 'amazon-cognito-auth-js';
 import { parse } from 'url';
+import { default as urlListener } from './urlListener';
 
 const logger = new Logger('AuthClass');
 const dispatchAuthEvent = (event, data) => {
     Hub.dispatch('auth', { event, data }, 'Auth');
-};
-
-let linkingHandlers = [];
-
-const urlListener = async (callback: ({ url: string }) => void) => {
-    if (Platform.isReactNative) {
-        let Linking: any;
-
-        try {
-            ({ Linking } = require('react-native'));
-        } catch (error) { /* Keep webpack happy */ }
-
-        const handler = ({ url, ...rest }: { url: string }) => {
-            logger.debug('addEventListener', { url, ...rest });
-            callback({ url });
-        };
-        linkingHandlers.forEach(lh => Linking.removeEventListener('url', lh));
-        Linking.addEventListener('url', handler);
-        linkingHandlers = [handler];
-
-        const initialUrl = await Linking.getInitialURL();
-        logger.debug('before callback', { initialUrl });
-        callback({ url: initialUrl });
-    } else if (JS.browserOrNode().isBrowser && window.location) {
-        const url = window.location.href;
-
-        callback({ url });
-    } else {
-        throw new Error('Not supported');
-    }
 };
 
 export enum CognitoHostedUIIdentityProvider {

--- a/packages/auth/src/urlListener.native.ts
+++ b/packages/auth/src/urlListener.native.ts
@@ -1,0 +1,38 @@
+/*
+ * Copyright 2017-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+import {
+    ConsoleLogger as Logger
+} from '@aws-amplify/core';
+const logger = new Logger('urlListener');
+
+let linkingHandlers = [];
+
+export default async callback => {
+    let Linking: any;
+
+    try {
+        ({ Linking } = require('react-native'));
+    } catch (error) { /* Keep webpack happy */ }
+
+    const handler = ({ url, ...rest }: { url: string }) => {
+        logger.debug('addEventListener', { url, ...rest });
+        callback({ url });
+    };
+    linkingHandlers.forEach(lh => Linking.removeEventListener('url', lh));
+    Linking.addEventListener('url', handler);
+    linkingHandlers = [handler];
+
+    const initialUrl = await Linking.getInitialURL();
+    logger.debug('before callback', { initialUrl });
+    callback({ url: initialUrl });
+};

--- a/packages/auth/src/urlListener.ts
+++ b/packages/auth/src/urlListener.ts
@@ -1,0 +1,25 @@
+/*
+ * Copyright 2017-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+import {
+    JS,
+} from '@aws-amplify/core';
+
+export default callback => {
+    if (JS.browserOrNode().isBrowser && window.location) {
+        const url = window.location.href;
+
+        callback({ url });
+    } else {
+        throw new Error('Not supported');
+    }
+};

